merino/providers/weather/backends/accuweather.py

```diff
@@ -92,18 +94,26 @@ async def get_weather_report(
         async with AsyncClient(base_url=self.url_base) as client:
             if not (location := await self.get_location(client, country, postal_code)):
                 return None
+            try:
+                async with asyncio.TaskGroup() as tg:
+                    task_current = tg.create_task(
+                        self.get_current_conditions(client, location.key)
+                    )
+                    task_forecast = tg.create_task(
+                        self.get_forecast(client, location.key)
+                    )
+            except BaseExceptionGroup as e:
+                raise AccuweatherError(f"err:{e.exceptions}")
 
-            if not (current := await self.get_current_conditions(client, location.key)):
-                return None
-
-            if not (forecast := await self.get_forecast(client, location.key)):
-                return None
-
-        return WeatherReport(
-            city_name=location.localized_name,
-            current_conditions=current,
-            forecast=forecast,
-        )
+            current = await task_current
+            forecast = await task_forecast
+            if current and forecast:
```

Comment:
Is this allowed?
```suggestion
if (current := await task_current) and (forecast := await task_forecast):
```

What was the verdict on _weakref_? Does it apply?

Reply:
Very snazzy Pythonic approach @Trinaa 👏

Reply:
I think it is. We can do:
```py
    return (
        WeatherReport(
            city_name=location.localized_name,
            current_conditions=current,
            forecast=forecast,
        )
        if (current := await task_current) and (forecast := await task_forecast)
        else None
    )
```